### DISCUSSION TITLE
Cyclic loading

### DISCUSSION THREAD

**Discussion by engrmessi** [idaholab/moose#28312] [Q&A Modules: General]
Good afternoon

I successfully simulated phase field model of diffusion controlled stress corrosion cracking SCC. In this model, my loading conditions was in form of strain rate, i.e. I applied strain rate (time displacement) at the right boundary as shown in the section of my code share here. Now my Advisor wants me to change the loading condition with cyclic loading to simulate fatigue cracking. Could anyone help me with how to implement cyclic load on BC block. Thank you
![help_Screenshot 2024-08-05 142203](https://github.com/user-attachments/assets/abf14f2c-3f61-4abf-825c-71a68243dc8f)

**Comment by GiudGiud** [comment]
Hello

You can write down a cyclic load with for example a cosine function?
```
expression = 2.94e-7 * cos(2*pi * t / period)
symbol_names = 'period'
symbol_values = ...
```

**Comment by engrmessi** [comment]
Thank you for your response. 
I particularly want to know the type of function to use. Is it still 'ParsedFunction'? or Wound "FunctionExpression" be a better option?
Is there any MOOSE documentation for cyclic loading, if yes, please share the link with me. I already searched, but I could not find any

Best regards

**Reply by GiudGiud** [reply]
parsedfunction can be great to create the period loading.
you can also chain functions if you want a complex profile in one and the periodicity in the other

**Reply by engrmessi** [reply]
Thank you for your feedback. I am still not successful

[Functions]
  [tdisp]
    type = ParsedFunction
    expression = 2.94e-7 * cos(2*pi*t/period) 
    symbol_names = 'period'
    symbol_values = '16'
  []
[]
Above is what I incorporated in my script, but my simulation is failing. the error says "syntax error, unexpected invalid token, expecting = or :=" 
I also used the below function and the error persists
[Functions]
  [tdisp]
    type = ParsedFunction
    expression = 2.94e-7 * cos(2*pi*t/16) 
  []
[]
Please how do I fix this syntax error. This error occurs in the line containing   expression = 2.94e-7 * cos(2*pi*t/period)  and expression = 2.94e-7 * cos(2*pi*t/16)

**Reply by engrmessi** [reply]
Sorry, I fixed the syntax problem. 

Best regards

**Reply by GiudGiud** [reply]
Awesome. Yes you were simply missing multiply symbols

**Comment by engrmessi** [comment]
Good morning GiudGiud, 
Still on modeling stress cracking fatigue under cyclic load. My PI asked that I model two cases (1) strain controlled cyclic load and (2) stress controlled cyclic load. 
The figures (a) and (b) are the input file and result (plot of stress vs time) obtain from strain controlled load which looks nice as the the peak stress is uniform. 
The figure (c) and (d) are for the stress controlled input file and plot of stress vs time. Here the pick stress is not uniform. My question is if this plot is ok. Must the peak stress be uniform as in the case (1). If this is not ok, what can I do to achieve the even stress pick?

Please also note that the amplitude of the functions used differs for both cases. case (1), requires 0.05 amplitude for the material to degrade while case (2) needs 50 amplitude, please see the input files used
Case (1): Strain controlled cyclic load
![(a) input file](https://github.com/user-attachments/assets/92302036-31f0-4218-840e-fcceb642bf69)
![(b) Vonmisses field](https://github.com/user-attachments/assets/f829f035-fa8f-4bbf-904e-792c78701c8e)
 
Case (2): Stress controlled cyclic load
![(c) input file](https://github.com/user-attachments/assets/0b9d0499-17de-44ea-ad6e-499fe14da0d0)
![(d) Vonmisses field](https://github.com/user-attachments/assets/a9911026-117a-4c24-a52b-cb7bb59a2dd8)

**Reply by GiudGiud** [reply]
Are these results converged in the time grid?

@bwspenc who would be a good person to comment on this?

**Comment by engrmessi** [comment]
Yes the results converged, however because there are a lot oscillations (cycle time) ....... up to 200000s, I only plot up to just 1200s and 2000s respectively.

**Comment by engrmessi** [comment]
These are the fatigue cracking for the cases respectively
![phase field variable](https://github.com/user-attachments/assets/0f6f9312-7ce3-4842-a7e4-760bbf834bfe)
![phase field](https://github.com/user-attachments/assets/56a42191-5920-48eb-bf5b-890dd80a763e)
